pkg/loop/loopinstall/main.go

```diff
@@ -0,0 +1,621 @@
+
+		if goPrivate != "" && envGoPrivate != "" {
+			// Append with comma separator
+			goPrivate = goPrivate + "," + envGoPrivate
+		} else if goPrivate == "" && envGoPrivate != "" {
+			// Just use the environment variable
+			goPrivate = envGoPrivate
+		}
+
+		os.Setenv("GOPRIVATE", goPrivate)
+		defer os.Setenv("GOPRIVATE", origGoPrivate)
+	}
+
+	// Expand environment variables
+	moduleURI := expandEnvVars(plugin.ModuleURI)
+	gitRef := expandEnvVars(plugin.GitRef)
+	installPath := expandEnvVars(plugin.InstallPath)
+
+	// Validate inputs
+	if err := validateModuleURI(moduleURI); err != nil {
+		return fmt.Errorf("validation failed: %w", err)
+	}
+
+	if gitRef != "" {
+		if err := validateGitRef(gitRef); err != nil {
+			return fmt.Errorf("validation failed: %w", err)
+		}
+	}
+
+	if err := validateInstallPath(installPath); err != nil {
+		return fmt.Errorf("validation failed: %w", err)
+	}
+
+	// Clean the install path
+	cleanInstallPath := filepath.Clean(installPath)
+
+	// Full module path with git reference
+	fullModulePath := moduleURI
+	if gitRef != "" {
+		fullModulePath = fmt.Sprintf("%s@%s", moduleURI, gitRef)
+	}
+
+	log.Printf("Installing plugin %s[%d] from %s", pluginType, pluginIdx, fullModulePath)
+
+	// Download the module and get its directory
+	var moduleDir string
+	{
+		cmd := exec.Command("go", "mod", "download", "-json", fullModulePath)
+		var out bytes.Buffer
+		cmd.Stdout = &out
+		cmd.Stderr = os.Stderr
+
+		if err := execCommand(cmd); err != nil {
+			return fmt.Errorf("failed to download module %s: %w", fullModulePath, err)
+		}
+
+		var result ModDownloadResult
+		if err := json.Unmarshal(out.Bytes(), &result); err != nil {
+			return fmt.Errorf("failed to parse go mod download output: %w", err)
+		}
+
+		moduleDir = result.Dir
+		if moduleDir == "" {
+			return fmt.Errorf("empty module directory returned for %s", fullModulePath)
+		}
+	}
+
+	// Build goflags
+	goflags := defaults.GoFlags
+	if envGoFlags := os.Getenv("CL_PLUGIN_GOFLAGS"); envGoFlags != "" {
+		goflags = envGoFlags
+	}
+
+	// Validate goflags
+	if goflags != "" {
+		if err := validateGoFlags(goflags); err != nil {
+			return fmt.Errorf("validation failed: %w", err)
+		}
+	}
+
+	// Install the plugin
+	{
+		args := []string{"install"}
+		if goflags != "" {
+			args = append(args, strings.Fields(goflags)...)
+		}
+
+		// Add the install path
+		args = append(args, filepath.Join(cleanInstallPath))
+
+		cmd := exec.Command("go", args...)
+		cmd.Dir = moduleDir
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+
+		log.Printf("Running install command: go %s (in directory: %s)", strings.Join(args, " "), moduleDir)
+
+		if err := execCommand(cmd); err != nil {
+			return fmt.Errorf("failed to install plugin %s[%d]: %w", pluginType, pluginIdx, err)
+		}
+	}
+
+	return nil
+}
+
+// writeBuildManifest writes installation artifacts to the specified file
+func writeBuildManifest(tasks []PluginInstallTask, outputFile string) error {
```

Comment:
What's the point of the output `BuildManifest` file - isn't this the same data declared on the input as build metadata?